src/librustc_errors/diagnostic_builder.rs

```diff
@@ -135,13 +135,9 @@ impl<'a> DiagnosticBuilder<'a> {
         }
 
         let handler = self.0.handler;
+        let dummy = Diagnostic::new(Level::Help, "");
+        let diagnostic = std::mem::replace(&mut self.0.diagnostic, dummy);
```

Comment:
Shouldn't we do `self.cancel()` here, to prevent the dummy diagnostic from being emitted? And maybe the `Level` should be `Bug`, so that we catch if it does get emitted?

Reply:
`Diagnostic` does not have a `Drop` impl and we are consuming `self` by value so I don't see how it matters.

Reply:
I just tested this code out on the input consisting of a single closing closing brace:
```rust
}
```

and I get: "internal compiler error: the following error was constructed but not emitted"

Reply:
Hmm yes, good point. That is happening because `Drop` is called on `self`, so we do indeed have to do `self.cancel();`.

Reply:
Bleh, this code caused lots of test failures. I fixed it by changing the level to `Level::Cancelled`, which avoids the need for the explicit `self.cancel()` call. But I forgot to `git commit -a --amend`... I really wish git wouldn't let you push when uncommitted changes are present. I have now pushed that change.